File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/utils/UserContext.java
package uk.gov.justice.digital.hmpps.riskprofiler.utils;

import org.springframework.stereotype.Component;


@Component
public class UserContext {
    private static final ThreadLocal<String> authToken = new ThreadLocal<>();

    public static String getAuthToken() { return authToken.get(); }
    public static void setAuthToken(String aToken) {authToken.set(aToken);}
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/utils/UserContext.kt
package uk.gov.justice.digital.hmpps.riskprofiler.utils

import org.springframework.stereotype.Component

@Component
object UserContext {
  private val authToken_ = ThreadLocal<String>()

  @JvmStatic
  var authToken: String?
    get() = authToken_.get()
    set(aToken) {
      authToken_.set(aToken)
    }
}
